README.md

```diff
@@ -1,4 +1,76 @@
-# StudentAPI-nodeJs
-* A restful API that perform CRUD operations on Students' and courses' data
-* student can take or drop courses
-* Link: https://danielstudent-api.herokuapp.com/api
+# Student Web Server
+
+Student Web Server is a RESTful API that manages student's and courses' information the API performs CRUD operations on student and course information, and returns the results in JSON format. The API can allow students to register for courses, and students can view their current courses.
+
+## Table of Contents
+
+- [Motivation](#motivation)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Testing](#api-testing)
+- [Tools](#tools)
+- [Heroku](#heroku)
+- [Authors](#authors)
+
+# Motivation
+
+Student Web Server was inspired by my college schedule builer, and I wanted to create a web application that would allow student to create my their own schedule by adding courses to their schedule or removing courses from their schedule.
+
+# Installation
+
+Use the package manager npm to install Student Web Server.
+
+```bash
+npm install
+```
+
+# Usage
+
+Use the following commands to start the application.
+
+```bash
+npm start
+```
+
+Users are required set the environment variables for the application to work.
+eg.
+DB_CONNECTION for the database connection
+JWT_SECRET for the JWT token secret
+JWT_REFRESH_ACCESS_TOKEN for the JWT refresh token secret
+
+# Tools
+
+Student Web Server uses the following tools:
+Node.js, NPM, MongoDB, Mongoose, Express, Passport, JWT, Bcrypt, JOI
+
+# Heroku
+
+- [Heroku Link](https://danielstudent-api.herokuapp.com/api)
+
+# API Testing
+
+The API has test cases for all the endpoints.
+The API is tested using the following tools:
+JEST
+command for running the tests:
```

Comment:
oh cool! You should (later) on add a GitHub action to run these tests

Reply:
I'm hoping to but please try and review the test :)